### ISSUE TITLE
Support non-standard julia registries

### ISSUE THREAD

**Issue by davidanthoff** [jupyter/repo2docker#602]
For my specific binder use case it would be great if one could specify a list of alternative julia registries, not just the general registry, from which packages that are listed in `Project.toml` should be loaded.

I assume that at some point julia will add an ability that one can specify alternative registries in the `Project.toml` itself (see https://github.com/JuliaLang/Pkg.jl/issues/635), but until that is the case, we could add an option for a `.binderjuliaregistries` file that is simply a text file with URLs for registries that should be used in addition to the default one.

Pinging @KristofferC just to get his thoughts whether that is a bad idea for some reason :)

**Comment by betatim** [comment]
I am not super familiar with Julia packaging but this sounds similar to personal package archives (PPA) in Debian or Ubuntu. We currently have no support for those when it comes to installing APT packages. This is on purpose as there is no community standard for specifying them in a config file. As a policy we don't want to [invent new file formats](https://xkcd.com/927/) and instead prefer to wait for the respective communities to create something that can be adopted by repo2docker.

I'd apply the same reasoning to custom registries for Julia. Especially if support via `Project.toml` is on the horizon.

The people who have asked for PPA support are vocal but there aren't many of them. To them the recommendation is to use a `Dockerfile`. I think for customer Julia registries one of the simpler escape hatches built in to repo2docker would work ([`postBuild`](https://repo2docker.readthedocs.io/en/latest/config_files.html#postbuild-run-code-after-installing-the-environment)).

**Comment by davidanthoff** [comment]
Fair enough :) I think in terms of an escape hatch, we would need a way to run a script _before_ the stuff that runs as part of the `get_assemble_scripts` to work around this.
